src/db.rs

```diff
@@ -610,8 +620,11 @@ impl Db {
     }
 
     pub async fn flush(&self) -> Result<(), SlateDBError> {
-        self.inner.flush().await?;
-        self.inner.flush_memtables().await
+        if self.inner.wal_enabled() {
```

Comment:
This change was suggested by @rodesai and seemed reasonable to me. It currently causes `test_get_with_object_store_cache_stored_files` to fail I think because the expected manifest is written from `flush_memtables`.

Reply:
Ya, this change makes sense from a performance perspective. We're durable once the WAL is flushed if it's enabled. Question is whether we want `flush()` to semantically mean "data is durable" or "all state has been flushed" (including the manifest and memtables). Honestly, not sure. I suspect since `flush()` might be in performance sensitive areas we want just "data is durable". If so, then...

Question is, do we want to support some way to flush memtables, too? Or do we want to update that test? I don't feel like we should alter the code to accommodate one test, so I'm inclined the test should be updated so it doesn't fail.

Reply:
Yeah I'm also inclined to just update the test to reflect the behavior in this PR. I'm not sure what the use case would be for a slatedb user to request that the manifest/memtable be flushed to object store. Those are details internal to the db that the user doesn't really care about - they just want to know that their data is durably stored. I could see wanting the memtable (or at least the immutable memtables) to be flushed as part of _shutdown_ to avoid a heavy restore. That's a different use case though.

Reply:
Makes sense. I would also expect flush() to just ensure durability. I'll update the test.